lib/travis/api/v3/models/settings.rb

```diff
@@ -0,0 +1,29 @@
+module Travis::API::V3
+  class Models::Settings
+    attr_reader :repository
+
+    def initialize(repository)
+      @repository = repository
+    end
+
+    def to_h
+      defaults.merge(repository.settings || {})
+    end
+
+    def update(settings = {})
+      settings = defaults.merge(settings)
```

Comment:
Should this be `settings = to_h.merge(settings)` instead so that updating `builds_only_with_travis_yml` doesn't reset `maximum_number_of_builds` back to `0`?